src/programs/console_test/console_test.cpp

```diff
@@ -30,43 +30,60 @@
 class
 MyTestApp : public MyApp //public wxAppConsole
 {
-	wxString hiv_image_80x80x1_filename;
-	wxString hiv_images_80x80x10_filename;
-	wxString sine_wave_128x128x1_filename;
-	wxString numeric_text_filename;
-	wxString temp_directory;
+  wxString hiv_image_80x80x1_filename;
+  wxString hiv_images_80x80x10_filename;
+  wxString sine_wave_128x128x1_filename;
+  wxString numeric_text_filename;
+  wxString temp_directory;
 
-	public:
+  public:
     // We need DoCalculation so we can have a bool return type for automated testing and a noop DoInteractiveUserInput to allow it to run from the console.
-		bool DoCalculation();
+    bool DoCalculation();
     void DoInteractiveUserInput();
 
-		bool test_has_passed;
+    bool test_has_passed;
     bool all_tests_have_passed;
-
-		void TestMRCFunctions();
-		void TestFFTFunctions();
-		void TestScalingAndSizingFunctions();
-		void TestFilterFunctions();
-		void TestAlignmentFunctions();
-		void TestImageArithmeticFunctions();
-		void TestSpectrumBoxConvolution();
-		void TestImageLoopingAndAddressing();
-		void TestNumericTextFiles();
-		void TestClipIntoFourier();
-		void TestMaskCentralCross();
-		void TestStarToBinaryFileConversion();
+    /*
+      The tests build in complexity, such that the validity of anyone one test may depend on any number of the others.
+      Rather than rely on the tests being run in any given order, we instead, at the time the test is written,  
+      prescribe any functions used in that test should be checked against the list of tests that have passed.
+
+      When BeginTest() is called, the string of the test name is added to "test_results" and set to false, and current_test_name is set to test_name.
+      When EndTest() is called, the current_test_name is set to "test_has_passed".
+
+      To check the results, a list of strings is passed to the function CheckDependencies( { "Image::ClipIntoFourier", "Test Empirical Distribution" } ).
+      If the string test name is not found indicating a typo or the test is not run, or the value is false, the test fails.
+    */
+    std::unordered_map<std::string, bool> test_results;
+    bool CheckDependencies( std::initializer_list<std::string> list );
```

Comment:
The implementation is quite straightforward to me (almost feels like python), but I don't quite understand what this is checking for. Is the idea not to run a test if a test for a required function failed? Wouldn't it fail then anyway? Is that really necessary since we are aiming for all test to pass anyway?

Reply:
I guess from a CI checkpoint it doesn't make sense, that is pass/fail en toto as you say. I guess I'm thinking mainly from a debugging perspective...not chasing the wrong errors. 

It also is a mechanism to encourage more complete codebase coverage; although that means it is just the rule to add dep checks, rather than the checks themselves that are important.

I guess the bottom line is that dependencies checks should only ever fail after modification of console_tests, and are only indirectly related to CI?